File: GeneratedEntityReaderClasses/Components/IKLimbAttackerComponent.cs
namespace GeneratedNoitaClasses;
public class IKLimbAttackerComponent : NoitaComponentBase,  iNoitaType<IKLimbAttackerComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat radius { get; set; } = new();
	public NoitaFloat leg_velocity_coeff { get; set; } = new();
	public NoitaFloat targeting_radius { get; set; } = new();
	public NoitaBool targeting_raytrace { get; set; } = new();
	public NoitaString target_entities_with_tag { get; set; } = new();
	public NoitaCVector2<NoitaFloat> mTarget { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		radius.Read(s);
		leg_velocity_coeff.Read(s);
		targeting_radius.Read(s);
		targeting_raytrace.Read(s);
		target_entities_with_tag.Read(s);
		mTarget.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		radius.Write(s);
		leg_velocity_coeff.Write(s);
		targeting_radius.Write(s);
		targeting_raytrace.Write(s);
		target_entities_with_tag.Write(s);
		mTarget.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/IKLimbAttackerComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class IKLimbAttackerComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    radius:NoitaFloat = field(default_factory=NoitaFloat)
    leg_velocity_coeff:NoitaFloat = field(default_factory=NoitaFloat)
    targeting_radius:NoitaFloat = field(default_factory=NoitaFloat)
    targeting_raytrace:NoitaBool = field(default_factory=NoitaBool)
    target_entities_with_tag:NoitaString = field(default_factory=NoitaString)
    mTarget:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.radius.Read(s)
        self.leg_velocity_coeff.Read(s)
        self.targeting_radius.Read(s)
        self.targeting_raytrace.Read(s)
        self.target_entities_with_tag.Read(s)
        self.mTarget.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.radius.Write(s)
        self.leg_velocity_coeff.Write(s)
        self.targeting_radius.Write(s)
        self.targeting_raytrace.Write(s)
        self.target_entities_with_tag.Write(s)
        self.mTarget.Write(s)

